Automattic/studio
# Fix triggering themes and plugins fetch for the AI chat context

Related to https://github.com/Automattic/studio/pull/442#pullrequestreview-2233335976.

## Proposed Changes

- Update side effect logic within the `useChatContext` hook to ensure that themes and plugins are only retrieved when needed.

## Testing Instructions



**Preparation:**

- Before running the test, create a couple of sites.
- For each site, add different themes and plugins.

**Steps:**

- Apply the following patch. **It will display the chat context of the selected site in the AI Assistant tab.**
```patch
diff --git forkSrcPrefix/src/ipc-handlers.ts forkDstPrefix/src/ipc-handlers.ts
index 2d7973a52e11caca77c5214857de2ee50159cd96..c27b26f81d87086e364fc8f2e8ab2e8b529619fc 100644
--- forkSrcPrefix/src/ipc-handlers.ts
+++ forkDstPrefix/src/ipc-handlers.ts
@@ -612,6 +612,7 @@ export async function executeWPCLiInline(
 	if ( ! server ) {
 		throw new Error( 'Site not found.' );
 	}
+	console.log( '==== WP-CLI', siteId, args );
 	return server.executeWpCliCommand( args );
 }
 
diff --git forkSrcPrefix/src/components/content-tab-assistant.tsx forkDstPrefix/src/components/content-tab-assistant.tsx
index 1de1432a2943cc5861b2994ff9d199004a6f66d9..9a13ab06f4082fb47867efd95ea3e83b2d333114 100644
--- forkSrcPrefix/src/components/content-tab-assistant.tsx
+++ forkDstPrefix/src/components/content-tab-assistant.tsx
@@ -377,6 +377,12 @@ export function ContentTabAssistant( { selectedSite }: ContentTabAssistantProps
 
 	return (
 		<div className="relative min-h-full flex flex-col bg-gray-50">
+			Chat context:
+			<textarea
+				className="h-[380px] w-full"
+				readOnly
+				value={ JSON.stringify( currentSiteChatContext, null, 2 ) }
+			/>
 			<div
 				data-testid="assistant-chat"
 				className={ cx(

```
- Run the app with the command `npm start`.
- Wait until the app is loaded.
- Observe in the logs that two entries are displayed related to the WP-CLI command for initially selected site:
    - `==== WP-CLI <SELETED_SITE_ID> plugin list --format=json --status=active`
    - `==== WP-CLI <SELETED_SITE_ID> theme list --format=json`
- Navigate to the AI Assistant tab of the current selected site.
- Observe that the chat context includes the correct themes and plugins of the selected site.
- Select another site.
- Observe in the logs that two entries are displayed related to the WP-CLI command:
    - `==== WP-CLI <SELETED_SITE_ID> plugin list --format=json --status=active`
    - `==== WP-CLI <SELETED_SITE_ID> theme list --format=json`
- Observe that the chat context includes the correct themes and plugins of the selected site.
- Create a new site.
- Observe that no logs are shown before the new site is running.
- Wait until the site is created and running.
- Observe in the logs that two entries are displayed related to the WP-CLI command:
    - `==== WP-CLI <SELETED_SITE_ID> plugin list --format=json --status=active`
    - `==== WP-CLI <SELETED_SITE_ID> theme list --format=json`
- Observe that the chat context includes the correct themes and plugins of the selected site.

## Pre-merge Checklist
